src/commands/promote.ts

```diff
@@ -0,0 +1,112 @@
+import { MatrixClient, MessageEvent, MessageEventContent, MembershipEvent } from "matrix-bot-sdk"
+
+import { CommandError, sendMessage, sleep } from "src/utils"
+
+
+export const PROMOTE_COMMAND = "promote"
+export const POWER_LEVEL_ALIAS: Record<string, number> = {
+  default: 0,
+  moderator: 50,
+  admin: 100
+}
+const MATRIX_HOMESERVER = "matrix.parity.io"
+
+export async function runPromoteCommand(
+  roomId: string,
+  event: MessageEvent<MessageEventContent>,
+  args: string[],
+  client: MatrixClient,
+  botUserId: string,
+): Promise<string> {
+  // 1. Retrive and validate arguments
+  const [, userId, targetRoomId, powerLevelArg] = args
+  if (!userId || !userId.includes(`:${MATRIX_HOMESERVER}`)) {
+    const [, wrongHomeServer] = userId.split(":")
+    throw new CommandError(
+      `The provided user handle is not registered under ${MATRIX_HOMESERVER}, but ${wrongHomeServer}. \nMake sure that username ends with ":${MATRIX_HOMESERVER}"`,
+    )
+  }
+  if (!targetRoomId || !targetRoomId.includes(`:${MATRIX_HOMESERVER}`)) {
+    const [, wrongHomeServer] = targetRoomId.split(":")
+    throw new CommandError(
+      `The provided room handle is not registered under ${MATRIX_HOMESERVER}, but ${wrongHomeServer}. \nMake sure that the room handle ends with ":${MATRIX_HOMESERVER}"`,
+    )
+  }
+  if (!powerLevelArg) {
```

Comment:
so the `0` as powerLevelArg won't be accepted or it's a string?

Reply:
at this stage `powerLevelArg ` is a string argument of the command